core/nodejsActionBase/src/whisk.js

```diff
@@ -104,28 +104,42 @@ function Whisk(apihost, logger) {
         //
         // Project the appropriate value based on invoke mode and HTTP statusCode
         // where success is 200, and anything else is a failure.
-        post(packet, logger, function(error, response, body) {
-            logger && logger.info('[whisk]', '[invoke]', 'response body', body);
+        var postPromise = new Promise(function(resolve, reject) {
+          post(packet, logger, function(error, response, body) {
+              logger && logger.info('[whisk]', '[invoke]', 'response body', body);
 
-            var activation = undefined;
+              var activation = undefined;
 
-            error = error || undefined;
-            response = response || {};
-            body = body || {};
+              error = error || undefined;
+              response = response || {};
+              body = body || {};
 
-            if (response.statusCode == 200 || response.statusCode == 202) {
-                activation = {
-                   activationId: body.activationId, // id always present
-                   result: (body.response || {}).result // may not exist if non-blocking
-                };
-            } else if (!error) {
-                // activation failed, set error to API host error response.
-                error = body.error + ' (' + body.errorCode + ')';
-            }
+              if (response.statusCode == 200 || response.statusCode == 202) {
+                  activation = {
+                     activationId: body.activationId, // id always present
+                     result: (body.response || {}).result // may not exist if non-blocking
```

Comment:
This is tricky (and not necessarily a problem introduced by this revision): if we set `result` to `{}` when it's unavailable, how can the user differentiate between a blocking call that ended up returning before the result was available (a possibility documented in the REST API), and a blocking call that did return an empty object as the result?